### PR TITLE
PYTHON-4483 Use hatch as task runner

### PR DESCRIPTION
**Pull request by blink1073**

### CODE REVIEW

**Comment by NoahStapp** [comment]
Unrelated EG test failure? https://spruce.mongodb.com/task/mongo_python_driver_tests_windows_python_version__platform~windows_64_vsMulti_small_auth_ssl~auth_ssl_python_version_windows~3.9_test_rapid_replica_set_patch_5cba4c44fe9ec633232f08bf8c6086252a3e5573_668db265bb8f750007f12779_24_07_09_21_57_58?execution=1&sortBy=STATUS&sortDir=ASC

File: .evergreen/run-tests.sh [hunk 1]
```diff
@@ -123,9 +123,9 @@ if [ -n "$TEST_ENCRYPTION" ] || [ -n "$TEST_FLE_AZURE_AUTO" ] || [ -n "$TEST_FLE
 
     python -m pip install '.[encryption]'
 
-    # Install libmongocrypt if necessary.
+    # Setup encryption necessary.
```
Comment (NoahStapp) [inline .evergreen/run-tests.sh:126]:
```suggestion
    # Setup encryption if necessary.
```

File: .evergreen/teardown-encyrption.sh
Comment (NoahStapp) [inline .evergreen/teardown-encyrption.sh:1]: Typo in name, should be `teardown-encryption.sh`

File: hatch.toml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+[envs.lint.scripts]
+run = "pre-commit run --all-files"
+run-manual = "pre-commit run --all-files --hook-stage manual"
+
+[envs.test]
+features = ["test"]
+[envs.test.scripts]
+test = "pytest -v --durations=5 --maxfail=10 {args}"
+test-eg = "bash ./.evergreen/run-tests.sh {args}"
+test-async = "test test/asynchronous/ {args}"
```
Comment (NoahStapp) [inline hatch.toml:44]: Hatch lets you reference previously-defined commands? Nice.

**Review by NoahStapp — CHANGES_REQUESTED** [review]
Way more concise than tox, awesome! Just a few minor typos.

File: .evergreen/teardown-encyrption.sh
Comment (blink1073) [reply .evergreen/teardown-encyrption.sh:1]: Fixed!

**Review by ShaneHarvey — APPROVED** [review]

**Pull request merged**